src/pool/srv_pool.c

```diff
@@ -2732,6 +2732,172 @@ ds_pool_svc_update_acl(uuid_t pool_uuid, d_rank_list_t *ranks,
 	return rc;
 }
 
+/**
+ * Delete entries in a pool's ACL without having a handle for the pool
+ */
+void
+ds_pool_acl_delete_handler(crt_rpc_t *rpc)
+{
+	struct pool_acl_delete_in	*in = crt_req_get(rpc);
+	struct pool_acl_delete_out	*out = crt_reply_get(rpc);
+	struct pool_svc			*svc;
+	struct rdb_tx			tx;
+	int				rc;
+	daos_prop_t			*prop = NULL;
+	struct daos_prop_entry		*entry;
+
+	D_DEBUG(DF_DSMS, DF_UUID": processing rpc %p\n",
+		DP_UUID(in->pdi_op.pi_uuid), rpc);
+
+	rc = pool_svc_lookup_leader(in->pdi_op.pi_uuid, &svc,
+				    &out->pdo_op.po_hint);
+	if (rc != 0)
+		D_GOTO(out, rc);
+
+	rc = rdb_tx_begin(svc->ps_rsvc.s_db, svc->ps_rsvc.s_term, &tx);
+	if (rc != 0)
+		D_GOTO(out_svc, rc);
+
+	/*
+	 * We need to read the old ACL, modify, and rewrite it
+	 */
+	ABT_rwlock_wrlock(svc->ps_lock);
+
+	rc = pool_prop_read(&tx, svc, DAOS_PO_QUERY_PROP_ACL, &prop);
+	if (rc != 0)
+		D_GOTO(out_lock, rc);
+
+	entry = daos_prop_entry_get(prop, DAOS_PROP_PO_ACL);
+	if (entry == NULL) {
+		D_ERROR(DF_UUID": No ACL prop entry for pool\n",
+			DP_UUID(in->pdi_op.pi_uuid));
+		D_GOTO(out_prop, rc);
+	}
+
+	rc = daos_acl_remove_ace((struct daos_acl **)&entry->dpe_val_ptr,
+				 in->pdi_type, in->pdi_principal);
+	if (rc != 0) {
+		D_ERROR(DF_UUID": Failed to remove requested principal, "
+			"rc=%d\n", DP_UUID(in->pdi_op.pi_uuid), rc);
+		D_GOTO(out_prop, rc);
+	}
+
+	rc = pool_prop_write(&tx, &svc->ps_root, prop);
+	if (rc != 0) {
+		D_ERROR(DF_UUID": failed to write updated ACL for pool: %d\n",
+			DP_UUID(in->pdi_op.pi_uuid), rc);
+		D_GOTO(out_prop, rc);
+	}
+
+	rc = rdb_tx_commit(&tx);
+	if (rc != 0)
+		D_GOTO(out_prop, rc);
```

Comment:
these 2 lines are unnecessary in the current function's structure, since the execution will flow through to out_prop either way.

Reply:
Good catch - I've seen mixed usage on this sort of thing. The goal being, I think, to minimize error if someone decides to add onto the function in the future. I can remove it, as it doesn't seem likely this function will be heavily modified.